### PR TITLE
New & fixed examples for aws_ssm lookup

### PR DESCRIPTION
**Pull request by kfattig**
##### SUMMARY
Updated examples based on discussion [here](https://github.com/ansible/ansible/issues/57862)

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
aws_ssm lookup

##### ADDITIONAL INFORMATION
Removed a non-working example, replaced with two working examples

### CODE REVIEW

File: lib/ansible/plugins/lookup/aws_ssm.py [hunk 1]
```diff
@@ -96,9 +96,13 @@
...
   debug: msg="{{ lookup('aws_ssm', '/PATH/to/params', region='ap-southeast-2', shortnames=true, bypath=true, recursive=true ) }}"
 
-- name: Iterate over a parameter hierarchy
-  debug: msg='key contains {{item.Name}} with value {{item.Value}} '
-  loop: '{{ query("aws_ssm", "/TEST/test-list", region="ap-southeast-2", bypath=true) }}'
+- name: Iterate over a parameter hierarchy (one iteration per parameter)
+  debug: msg='Key contains {{ item.key }} , with value {{ item.value }}'
+  loop: '{{ lookup("aws_ssm", "/demo/", bypath=True) | dict2items }}'
+
+- name: Iterate over multiple paths (one iteration per path)
```
Comment (s-hertel) [inline lib/ansible/plugins/lookup/aws_ssm.py]: Looks good. Perhaps you could clarify that this is iterating over a list of dictionaries.

File: lib/ansible/plugins/lookup/aws_ssm.py [hunk 2]
```diff
@@ -96,9 +96,13 @@
 - name: return a dictionary of ssm parameters from a hierarchy path with shortened names (param instead of /PATH/to/param)
   debug: msg="{{ lookup('aws_ssm', '/PATH/to/params', region='ap-southeast-2', shortnames=true, bypath=true, recursive=true ) }}"
 
-- name: Iterate over a parameter hierarchy
-  debug: msg='key contains {{item.Name}} with value {{item.Value}} '
-  loop: '{{ query("aws_ssm", "/TEST/test-list", region="ap-southeast-2", bypath=true) }}'
+- name: Iterate over a parameter hierarchy (one iteration per parameter)
+  debug: msg='Key contains {{ item.key }} , with value {{ item.value }}'
+  loop: '{{ lookup("aws_ssm", "/demo/", bypath=True) | dict2items }}'
```
Comment (s-hertel) [inline lib/ansible/plugins/lookup/aws_ssm.py]: Also no need to remove region from the example.
```suggestion
  loop: '{{ lookup("aws_ssm", "/demo/", region="ap-southeast-2", bypath=True) | dict2items }}'
```

File: lib/ansible/plugins/lookup/aws_ssm.py (hunk 1 above)
Comment (kfattig) [reply lib/ansible/plugins/lookup/aws_ssm.py]:
```suggestion
- name: Iterate over multiple paths as dictionaries (one iteration per path)
```

**Review by s-hertel — APPROVED** [review]

**Comment by acozine** [comment]
Thanks @kfattig for turning a conversation into usable examples in the documentation!

**Pull request merged**